### ISSUE TITLE
Undefined method blank? in dynadns library

### ISSUE THREAD

**Issue by russellseymour** [chef-brigade/dhcp-cookbook#67]
I am trying to use the latest version of the DHCP cookbook and am getting this error:

```
  NoMethodError
  -------------
  undefined method `blank?' for "127.0.0.1":String

  Relevant File Content:
  ----------------------
  /var/chef/cache/cookbooks/dhcp/libraries/dynadns.rb:

   26:          masters ||= {}
   27:          return unless @zones
   28:          @zones.each do |zone|
   29:            name = zone['zone_name']
   30:            masters[name] ||= {}
   31:
   32:            # set to global master by default
   33>>           if node['dns'].key?(:master) && node['dns']['master'].blank? == false
   34:              masters[name]['master'] = node['dns']['master']
   35:            end
   36:
   37:            if node['dns'].key?(:rndc_key) && node['dns']['rndc_key'].blank? == false
   38:              masters[name]['key'] = node['dns']['rndc_key']
   39:            end
   40:
   41:            # use zone bag override if it exists
   42:            if zone.key?('master_address') && zone['master_address'].blank? == false
```

The relevant attributes in my wrapper cookbook are:

```
default[:dns][:master] = "127.0.0.1"
default[:dns][:rndc_key] = "rndc-key"
```

I am not really sure what the error is here as I have been using an older version of the cookbook (very old in fact 2.1.1) and this was not an issue.  The versions of software I am using are as follows:

Ubuntu: 14.04.3
Ruby: 2.2.4
Chef: 12.6.0

I can only think it is do with the versions somehow or I am missing something obvious.  Any assistance gratefully received.

Thanks, Russell

**Comment by jmccann** [comment]
Thanks for bringing up this issue Russell.  I'll take a look.

**Comment by spheromak** [comment]
@jmccann you removed the ruby-helpers which had implemented blank?  I believe this is also implemented in chef-sugar now as well.

**Comment by spheromak** [comment]
if you were wanting to keep blank, but not wanting my unmaintained helpers, using chef-sugar and `require 'chef/sugar/core_extensions'` would bring  `Object.blank?` back

**Comment by jmccann** [comment]
Thanks @spheromak .  I'll look into that

Figured out why spec testing continues to pass ... because I'm stupid and added https://github.com/chef-brigade/dhcp-cookbook/blob/master/spec/spec_helper.rb#L11-L16

My plan (time continuing to permit) is to address and release a fix tomorrow.

**Comment by jmccann** [comment]
@russellseymour Thanks again for bringing this up.  I got to this sooner then expected and released 5.3.2 with a bug fix for this.

Let me know if you continue to have issues with this or anything else.

Thanks!
